shm/inc/Island.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+class Island {
+public:
+    class Coordinates {
+    public:
+        Coordinates(int, int);
+        bool operator==(const Coordinates&);
```

Comment:
Method must be const from C++20 to avoid warning about ambiguous reversed operator.

bool operator==(/* this*/ A&, A const&); // member function
bool operator==(A const&, /* this*/ A&); // reversed member function

With two arguments of type A. The first candidate is better in the first argument, and the second candidate is better in the second argument. Neither candidate is better than the other, hence ambiguous.